### PR TITLE
Add models for Spring validation.Errors

### PR DESCRIPTION
**Pull request by sauyon**
Far less work than one would imagine.

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/spring/SpringErrors.qll [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+private class SpringErrorModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "org.springframework.validation;Errors;true;addAllErrors;;;Argument[0];Argument[-1];taint",
+        "org.springframework.validation;Errors;true;getAllErrors;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.validation;Errors;true;getFieldError;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.validation;Errors;true;getFieldErrors;;;Argument[-1];ReturnValue;taint",
+        // for review: excluding because I think field values are unlikely to carry meaningful taint, though I may be wrong.
```
Comment (smowton) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringErrors.qll:12]: It's not clear just from the interface where we'd propagate taint from (`this`, or some associated object chosen by the implementation?), so I'm inclined to agree

**Comment by smowton** [comment]
```
Error: The following CodeQL elements are lacking documentation:
semmle/code/java/frameworks/spring/SpringErrors.qll  SpringErrors  file
```

File: java/ql/test/library-tests/frameworks/spring/ValidationErrorsTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import org.springframework.validation.Errors;
```
Comment (aschackmull) [inline java/ql/test/library-tests/frameworks/spring/ValidationErrorsTest.java:1]: Looks like you forgot to commit a few files: the `.ql` and `.expected` files are missing here.

File: java/ql/src/semmle/code/java/frameworks/spring/SpringErrors.qll [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/** Definitions of flow steps through utility methods of `org.springframework.validation.Errors`. */
+
+import java
+private import semmle.code.java.dataflow.ExternalFlow
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringErrors.qll:4]: This needs bidirectional import. E.g. by importing `Spring.qll` in `ExternalFlow.qll`.

File: java/ql/test/stubs/springframework-5.2.3/org/springframework/validation/FieldError.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ * limitations under the License.
+ */
+
+package org.springframework.validation;
+
+import org.springframework.lang.Nullable;
+
+public class FieldError extends ObjectError {
+
+  private final String field;
```
Comment (aschackmull) [inline java/ql/test/stubs/springframework-5.2.3/org/springframework/validation/FieldError.java:23]: This file isn't quite just stubs. Get rid of superfluous code such as private fields and constructor bodies.

File: java/ql/test/stubs/springframework-5.2.3/org/springframework/validation/ObjectError.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.lang.Nullable;
+
+public class ObjectError extends DefaultMessageSourceResolvable {
+
+  private final String objectName;
+
+  @Nullable private transient Object source;
+
+  public ObjectError(String objectName, String defaultMessage) {
+    super("");
```
Comment (aschackmull) [inline java/ql/test/stubs/springframework-5.2.3/org/springframework/validation/ObjectError.java:29]: Remove code, this should just be stubs.

**Review by aschackmull — COMMENTED** [review]
There are a few empty stub files added. Remove?

**Comment by sauyon** [comment]
I didn't notice that this was waiting for me, I've applied the review comments.

**Review by aschackmull — APPROVED** [review]

**Pull request merged**